scout/adapter/mongo/case_events.py

```diff
@@ -546,36 +545,34 @@ def unshare(self, institute, case, collaborator_id, user, link):
         LOG.debug("Case updated")
         return updated_case
 
-    def diagnose(self, institute, case, user, link, omim_obj, omim_inds=[], remove=False):
-        """Diagnose a case using OMIM ids.
+    def diagnose(
+        self,
+        institute: dict,
+        case: dict,
+        user: dict,
+        link: str,
+        disease_id: str,
+        omim_inds: list = [],
+        remove: bool = False,
+    ) -> Optional[dict]:
+        """Add or remove a diagnose to a case and eventually case individuals."""
 
-        Args:
-            institute (dict): A Institute object
-            case (dict): Case object
-            user (dict): A User object
-            link (str): The url to be used in the event
-            omim_obj(dict): An OMIM term dictionary
-            omim_inds(list): List of case individuals with diagnosis
-            remove(bool):
-
-        Return:
-            updated_case
-
-        """
-        omim_modif_id = omim_obj["_id"]  # OMIM ID to add or remove from case diagnoses
         updated_diagnoses = []
-        case_diagnoses = case.get("diagnosis_phenotypes") or []
+        case_diagnoses = case.get("diagnosis_phenotypes", [])
```

Comment:
I don't think it would ever happen, given that when a case is loaded it doesn't have the key, and when you change it gets set to a list (even an empty one), but safe code is good, I'll fix!